testsuite/unittests/test_gif_cond_exp.sli

```diff
@@ -0,0 +1,112 @@
+/*
+ *  test_gif_cond_exp.sli
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+ /* BeginDocumentation
+Name: testsuite::test_gif_cond_exp - sli script for test of gif_cond_exp with external DC current and spike generators
+
+Synopsis: (test_gif_cond_exp) run
+
+
+Description:
+
+ This testscript checks the correctness of gif_psc_exp neuron model. It injects external DC current and also three spikes to two neurons and measures their spike times. The spike times should match the expected values.
+
+Author:  March 2016, Setareh
+SeeAlso: gif_cond_exp, testsuite::test_gif_cond_exp_multisynapse
+*/
```

Comment:
@hesam-setareh The test is fine, but I would suggest to bring it closer to usual code structure for tests and to avoid code duplication. You are doing exactly the same thing twice, except that the parameters for the model differ and that you expect different spike times. Therefore, it would be useful to write a function that runs the test for given parameters and just returns the spike times array (the `:` at the beginning of the name indicates a "private" function in SLI):
```
/:run_test
{
  /params Set 

  % build and run network, use this to create the neuron
  /gif_cond_exp params Create /neuron Set

 % connect and simulate, leave spike times on stack at end
} def
```
Then put each test in a separate block:

% Test 1: default parameters:
{
  ResetKernel    % each test should start with this
  << >> :run_test
  [ 426 849 1227 ] eq
} assert_or_die
```
and similar for the non-default case.

You may also want to add a few tests that check that parameter values are error-checked properly, e.g., prohibiting negative values:
```
{
  /gif_cond_exp << /lambda_0 -10. >> Create
} fail_or_die
```